modules/fledgeForGpt.js

```diff
@@ -48,5 +47,50 @@ export function addComponentAuctionHook(next, bidRequest, componentAuctionConfig
     logWarn(MODULE, `unable to register component auction config for: ${adUnitCode} x ${seller}.`);
   }
 
-  next(bidRequest, componentAuctionConfig);
+  next(adUnitCode, componentAuctionConfig);
 }
+
+export function markBidsForFledge(next, bidderRequests) {
+  if ('runAdAuction' in navigator && 'joinAdInterestGroup' in navigator) {
+    bidderRequests.forEach((req) => {
+      req.fledgeEnabled = config.runWithBidder(req.bidderCode, () => config.getConfig('fledgeEnabled'))
+    })
+  }
+  next(bidderRequests);
+}
+getHook('makeBidRequests').after(markBidsForFledge);
+
+export function setImpExtAe(imp, bidRequest, context) {
+  if (!context.bidderRequest.fledgeEnabled) {
+    delete imp.ext?.ae;
+  }
+}
+registerOrtbProcessor({type: IMP, name: 'impExtAe', fn: setImpExtAe});
+
+// to make it easier to share code between the PBS adapter and adapters whose backend is PBS, break up
+// fledge response processing in two steps: first aggregate all the auction configs by their imp...
+
+export function parseExtPrebidFledge(response, ortbResponse, context) {
+  (ortbResponse.ext?.prebid?.fledge?.auctionconfigs || []).forEach((cfg) => {
+    const impCtx = context.impContext[cfg.impid];
+    if (!impCtx?.imp?.ext?.ae) {
+      logWarn('Received fledge auction configuration for an impression that was not in the request or did not ask for it', cfg, impCtx?.imp);
```

Comment:
validation behavior is different between PBS and regular adapter. If adapters adopt this method it may become a non-issue but it may be confusing as is.

Reply:
I am not sure this validation makes sense outside of here. There are two things we are checking here - 

1. the request should have set `imp.ext.ae`; does this check make sense for an adapter that (in general) may not be using ORTB at all? 
2. the fledge config should refer to an imp that existed in the request. this is necessary for PBS because the protocol pairs a fledge config with `impid`. But, for client-side adapters,  they must provide a valid `bidId` instead - [the check is here](https://github.com/prebid/Prebid.js/blob/c05bba2920ebe3a97b231e058acc2c09db27bf10/src/adapters/bidderFactory.js#L261), although there's no warning.

I can add a warning for 2, I'm not sure about 1.

Reply:
Should be fine to leave it as it is now and update in a later iteration if it becomes a problem with usage.